third_party/move/move-bytecode-verifier/src/code_unit_verifier.rs

```diff
@@ -66,6 +290,10 @@ impl<'a> CodeUnitVerifier<'a> {
             )
             .map_err(|err| err.at_index(IndexKind::FunctionDefinition, index.0))?;
             total_back_edges += num_back_edges;
+            // check whether struct APIs related code is well-formed.
+            let resolver = BinaryIndexedView::Module(module);
+            Self::check_struct_api_impl(&resolver, module, function_definition)
+                .map_err(|err| err.at_index(IndexKind::FunctionDefinition, index.0))?;
```

Comment:
Document what prevents someone from adding these attributes to various functions for a struct not owned by that module (or do we need this check somewhere)?